tcl/org.testeditor.tcl.dsl.ide/src/org/testeditor/tcl/dsl/ide/TclContentProposalProvider.xtend

```diff
@@ -0,0 +1,296 @@
+package org.testeditor.tcl.dsl.ide
+
+import com.google.common.base.Predicate
+import javax.inject.Inject
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.xtext.CrossReference
+import org.eclipse.xtext.EcoreUtil2
+import org.eclipse.xtext.RuleCall
+import org.eclipse.xtext.ide.editor.contentassist.ContentAssistContext
+import org.eclipse.xtext.ide.editor.contentassist.IIdeContentProposalAcceptor
+import org.eclipse.xtext.ide.editor.contentassist.IdeContentProposalProvider
+import org.eclipse.xtext.resource.IEObjectDescription
+import org.eclipse.xtext.xtext.CurrentTypeFinder
+import org.slf4j.LoggerFactory
+import org.testeditor.aml.Component
+import org.testeditor.aml.ComponentElement
+import org.testeditor.aml.InteractionType
+import org.testeditor.aml.ModelUtil
+import org.testeditor.aml.Template
+import org.testeditor.aml.TemplateContainer
+import org.testeditor.aml.TemplateText
+import org.testeditor.aml.TemplateVariable
+import org.testeditor.dsl.common.util.JvmTypeReferenceUtil
+import org.testeditor.tcl.ComponentTestStepContext
+import org.testeditor.tcl.EnvironmentVariable
+import org.testeditor.tcl.MacroCollection
+import org.testeditor.tcl.MacroTestStepContext
+import org.testeditor.tcl.StepContentElement
+import org.testeditor.tcl.TestStep
+import org.testeditor.tcl.VariableReferencePathAccess
+import org.testeditor.tcl.dsl.jvmmodel.SimpleTypeComputer
+import org.testeditor.tcl.dsl.services.TclGrammarAccess
+import org.testeditor.tcl.util.TclModelUtil
+import org.testeditor.tsl.StepContent
+
+class TclContentProposalProvider extends IdeContentProposalProvider {
+
+	static val logger = LoggerFactory.getLogger(TclContentProposalProvider)
+	
+	@Inject JvmTypeReferenceUtil typeUtil
+	@Inject SimpleTypeComputer typeComputer
+	@Inject extension TclGrammarAccess
+	@Inject extension TclModelUtil
+	@Inject extension ModelUtil
+	@Inject CurrentTypeFinder currentTypeFinder
+
+	/**
+	 * hook into proposal construction and identify relevant cases
+	 */
+	override protected _createProposals(RuleCall ruleCall, ContentAssistContext context, IIdeContentProposalAcceptor acceptor) {
+		switch (ruleCall.rule) {
+			case testStepRule:
+				makeTestStepProposals(context, acceptor)
+			case stepContentElementRule:
+				makeStepContentElementProposals(context, acceptor)
+			case IDRule:
+				makeIDProposals(context, acceptor)
+			default: {
+			} // ignore
```

Comment:
maybe add a brief comment as to why anything else can be safely ignored, instead (because the fact that any other cases _are_ ignored is fairly obvious…)